### PR TITLE
Selfie: Implement saving location

### LINKED ISSUE #46: GPS location tagging

### PR DESCRIPTION
**Pull request by luk1337**
Change-Id: I37ddfc34eb037bcbfa564b8552fba6ecb89e2a8d

### CODE REVIEW

File: app/src/main/java/org/lineageos/selfie/SharedPreferencesExt.kt [hunk 1]
```diff
@@ -141,6 +141,18 @@ internal var SharedPreferences.photoCaptureMode: Int
         }
     }
 
+private const val PHOTO_SAVE_LOCATION = "photo_save_location"
+private const val PHOTO_SAVE_LOCATION_DEFAULT = false
+internal var SharedPreferences.photoSaveLocation: Boolean
```
Comment (luca020400) [inline app/src/main/java/org/lineageos/selfie/SharedPreferencesExt.kt]: This is ignored at the moment

File: app/src/main/java/org/lineageos/selfie/MainActivity.kt [hunk 2]
```diff
@@ -304,6 +342,8 @@ class MainActivity : AppCompatActivity() {
                     Toast.LENGTH_SHORT
                 ).show()
                 finish()
+            } else if (!allLocationPermissionsGranted()) {
+                ActivityCompat.requestPermissions(this, REQUIRED_PERMISSIONS_LOCATION, 0)
```
Comment (luca020400) [inline app/src/main/java/org/lineageos/selfie/MainActivity.kt]: No code for GPS?

**Review by luca020400 — COMMENTED** [review]
Please pull in https://github.com/luca020400/android_packages_apps_Selfie/commit/5e3ccb161301cf99541142319e0ad0809cdf987e
Should be split in add location to video and avoid adding null location

File: app/src/main/java/org/lineageos/selfie/MainActivity.kt (hunk 2 above)
Comment (luk1337) [reply app/src/main/java/org/lineageos/selfie/MainActivity.kt]: image

Comment (luca020400) [reply app/src/main/java/org/lineageos/selfie/MainActivity.kt]: Ayy [1]

File: app/src/main/java/org/lineageos/selfie/MainActivity.kt [hunk 3]
```diff
@@ -333,6 +373,7 @@ class MainActivity : AppCompatActivity() {
 
         // Create output options object which contains file + metadata
         val outputOptions = StorageUtils.getPhotoMediaStoreOutputOptions(contentResolver)
+        outputOptions.metadata.location = location
```
Comment (luca020400) [inline app/src/main/java/org/lineageos/selfie/MainActivity.kt]: https://github.com/luca020400/android_packages_apps_Selfie/commit/5e3ccb161301cf99541142319e0ad0809cdf987e

**Review by luca020400 — CHANGES_REQUESTED** [review]

**Comment by luk1337** [comment]
I think this is now good to go.

**Review by luca020400 — APPROVED** [review]

**Comment by luca020400** [comment]
Need to check if we don't use the GPS provider how good location is
Because I'd really like to kill GPS itself and used passive/fused

**Pull request merged**